crates/sui-benchmark/src/lib.rs

```diff
@@ -1,6 +1,297 @@
+    quorum_driver::{QuorumDriver, QuorumDriverHandler, QuorumDriverMetrics},
+};
+use sui_json_rpc_types::{SuiCertifiedTransaction, SuiObjectRead, SuiTransactionEffects};
+use sui_network::default_mysten_network_config;
+use sui_sdk::SuiClient;
+use sui_types::{
+    base_types::ObjectID,
+    committee::{Committee, EpochId},
+    error::SuiError,
+    messages::{
+        CertifiedTransactionEffects, QuorumDriverRequestType, QuorumDriverResponse, Transaction,
+    },
+    object::{Object, ObjectRead},
+};
+use sui_types::{
+    base_types::ObjectRef,
+    crypto::AuthorityStrongQuorumSignInfo,
+    messages::{ExecuteTransactionRequestType, QuorumDriverRequest},
+    object::Owner,
+};
+use tracing::{error, info};
 
 pub mod drivers;
 pub mod util;
 pub mod workloads;
+
+/// A wrapper on execution results to accommodate different types of
+/// responses from LocalValidatorAggregatorProxy and FullNodeProxy
+#[allow(clippy::large_enum_variant)]
+pub enum ExecutionEffects {
+    CertifiedTransactionEffects(CertifiedTransactionEffects),
+    SuiTransactionEffects(SuiTransactionEffects),
+}
+
+impl ExecutionEffects {
+    pub fn mutated(&self) -> Vec<(ObjectRef, Owner)> {
+        match self {
+            ExecutionEffects::CertifiedTransactionEffects(certified_effects) => {
+                certified_effects.effects.mutated.clone()
+            }
+            ExecutionEffects::SuiTransactionEffects(sui_tx_effects) => sui_tx_effects
+                .mutated
+                .clone()
+                .into_iter()
+                .map(|refe| (refe.reference.to_object_ref(), refe.owner))
+                .collect(),
+        }
+    }
+
+    pub fn created(&self) -> Vec<(ObjectRef, Owner)> {
+        match self {
+            ExecutionEffects::CertifiedTransactionEffects(certified_effects) => {
+                certified_effects.effects.created.clone()
+            }
+            ExecutionEffects::SuiTransactionEffects(sui_tx_effects) => sui_tx_effects
+                .created
+                .clone()
+                .into_iter()
+                .map(|refe| (refe.reference.to_object_ref(), refe.owner))
+                .collect(),
+        }
+    }
+
+    pub fn quorum_sig(&self) -> Option<&AuthorityStrongQuorumSignInfo> {
+        match self {
+            ExecutionEffects::CertifiedTransactionEffects(certified_effects) => {
+                Some(&certified_effects.auth_signature)
+            }
+            ExecutionEffects::SuiTransactionEffects(_) => None,
+        }
+    }
+
+    pub fn gas_object(&self) -> (ObjectRef, Owner) {
+        match self {
+            ExecutionEffects::CertifiedTransactionEffects(certified_effects) => {
+                certified_effects.effects.gas_object
+            }
+            ExecutionEffects::SuiTransactionEffects(sui_tx_effects) => {
+                let refe = &sui_tx_effects.gas_object;
+                (refe.reference.to_object_ref(), refe.owner)
+            }
+        }
+    }
+}
+
+#[async_trait]
```

Comment:
This trait does not seem to have a direct relationship with benchmark and could be useful for other usecases. I wonder if there is a better place for this trait and its implementations i.e. maybe the crate where we keep QuorumDriver class?

Reply:
can't put it in `sui-core` (where quorum driver is) because it introduces extra dependencies. Not entirely sure about the best crate at the moment but open to suggestions!